contracts/coordinator/contracts/src/libs/LibEIP712Domain.sol

```diff
@@ -18,61 +18,43 @@
 
 pragma solidity ^0.5.5;
 
+import "@0x/contracts-exchange-libs/contracts/src/LibEIP712.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibEIP712ExchangeDomainConstants.sol";
 import "./LibConstants.sol";
 
 
+// solhint-disable var-name-mixedcase
 contract LibEIP712Domain is
-    LibConstants
+    LibConstants,
+    LibEIP712,
+    LibEIP712ExchangeDomainConstants
 {
 
-    // EIP191 header for EIP712 prefix
-    string constant internal EIP191_HEADER = "\x19\x01";
-
     // EIP712 Domain Name value for the Coordinator
     string constant internal EIP712_COORDINATOR_DOMAIN_NAME = "0x Protocol Coordinator";
 
     // EIP712 Domain Version value for the Coordinator
-    string constant internal EIP712_COORDINATOR_DOMAIN_VERSION = "1.0.0";
-
-    // EIP712 Domain Name value for the Exchange
-    string constant internal EIP712_EXCHANGE_DOMAIN_NAME = "0x Protocol";
-
-    // EIP712 Domain Version value for the Exchange
-    string constant internal EIP712_EXCHANGE_DOMAIN_VERSION = "2";
-
-    // Hash of the EIP712 Domain Separator Schema
-    bytes32 constant internal EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH = keccak256(abi.encodePacked(
-        "EIP712Domain(",
-        "string name,",
-        "string version,",
-        "address verifyingContract",
-        ")"
-    ));
+    string constant internal EIP712_COORDINATOR_DOMAIN_VERSION = "2.0.0";
 
     // Hash of the EIP712 Domain Separator data for the Coordinator
-    // solhint-disable-next-line var-name-mixedcase
     bytes32 public EIP712_COORDINATOR_DOMAIN_HASH;
 
     // Hash of the EIP712 Domain Separator data for the Exchange
-    // solhint-disable-next-line var-name-mixedcase
     bytes32 public EIP712_EXCHANGE_DOMAIN_HASH;
 
     constructor ()
         public
     {
-        EIP712_COORDINATOR_DOMAIN_HASH = keccak256(abi.encodePacked(
-            EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH,
-            keccak256(bytes(EIP712_COORDINATOR_DOMAIN_NAME)),
-            keccak256(bytes(EIP712_COORDINATOR_DOMAIN_VERSION)),
-            uint256(address(this))
-        ));
-
-        EIP712_EXCHANGE_DOMAIN_HASH = keccak256(abi.encodePacked(
-            EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH,
-            keccak256(bytes(EIP712_EXCHANGE_DOMAIN_NAME)),
-            keccak256(bytes(EIP712_EXCHANGE_DOMAIN_VERSION)),
-            uint256(address(EXCHANGE))
-        ));
+        EIP712_COORDINATOR_DOMAIN_HASH = hashEIP712Domain(
```

Comment:
Shouldn't these 2 calculations include the `chainId` in the hash?

Reply:
Ah I see. Feels a bit inconsistent that `verifyingContractAddress` is passed into `hashEIP712Domain` and `chainId` is read from state. I'd probably just add `chainId` as a param too.